src/datetime/mod.rs

```diff
@@ -585,6 +585,49 @@ impl DateTime<FixedOffset> {
     }
 }
 
+impl DateTime<Utc> {
+    /// Parses an RFC 2822 date and time string such as `Tue, 1 Jul 2003 10:52:37 +0200`,
```

Comment:
Nit: would like these doc strings to start with a single concise line describing the functionality, then an empty line, then a longer/more detailed description.

Reply:
I agree. These were copy and pasted from the original impl for FixedOffset at the time the patch was first authored - I don’t know if those impls have had their docs updated but I agree that it would be the correct way given the way rust docs are generated and displayed and given that it’s the defacto approach the community has adopted. 

I’ll update all the docs in this PR accordingly and squash the changes.

Reply:
Cool, yeah, if you update the docs, let's merge this!

Note: this PR targets main but main is now intended for 0.5.0 development. Since this is additive, we could merge it into 0.4.x to make it available sooner, so maybe rebase and change the PR's target branch?